pkg/reconciler/certificate/certificate.go

```diff
@@ -160,8 +163,14 @@ func (c *Reconciler) reconcile(ctx context.Context, knCert *v1alpha1.Certificate
 		knCert.Status.MarkReady()
 		knCert.Status.HTTP01Challenges = []v1alpha1.HTTP01Challenge{}
 	case cmCertReadyCondition.Status == cmmeta.ConditionFalse:
-		knCert.Status.MarkFailed(cmCertReadyCondition.Reason, cmCertReadyCondition.Message)
-		knCert.Status.HTTP01Challenges = []v1alpha1.HTTP01Challenge{}
+		if strings.EqualFold(cmCertReadyCondition.Reason, inProgressReason) {
+			knCert.Status.MarkNotReady(cmCertReadyCondition.Reason, cmCertReadyCondition.Message)
+		} else {
```

Comment:
Do we have the list of possible statuses, perhaps something else should go in `NotReady`?

Reply:
Added a list of possible statuses based on https://github.com/jetstack/cert-manager/blob/b7e83b53820e712e7cf6b8dce3e5a050f249da79/pkg/controller/certificates/sync.go#L130